### PR TITLE
Panzer: vector basis mass matrix

### PR DESCRIPTION
**Pull request by rppawlo**
## Motivation

Improve performance of mass matrix assembly tools for vector bases

### CODE REVIEW

File: packages/panzer/disc-fe/src/Panzer_IntrepidOrientation.cpp [hunk 1]
```diff
@@ -176,7 +176,7 @@ namespace panzer {
           vertices(connMgr.getConnectivity(lid),numVerticesPerCell);
 
         // This function call expects a view for the vertices, not the nodes
-        orts[lid] = Intrepid2::Orientation::getOrientation(cellTopo, vertices);
```
Comment (cgcgcg) [inline packages/panzer/disc-fe/src/Panzer_IntrepidOrientation.cpp:179]: Huh. How did this work before?

File: packages/panzer/disc-fe/src/Panzer_L2Projection.cpp [hunk 2]
```diff
@@ -151,106 +161,90 @@ namespace panzer {
...
 
-          for(const auto& i : offsets)
-            kOffsets_h(i) = offsets[i];
+        const int numBasisPoints = static_cast<int>(weightedBasis.extent(1));
+        if ( use_lumping ) {
+          Kokkos::parallel_for(num_cells_owned,KOKKOS_LAMBDA (const int& cell) {
+            double total_mass = 0.0, trace = 0.0;
 
-          Kokkos::deep_copy(kOffsets, kOffsets_h);
+            panzer::LocalOrdinal cLIDs[256];
```
Comment (cgcgcg) [inline packages/panzer/disc-fe/src/Panzer_L2Projection.cpp:190]: Define the magic 256 via macro?

File: packages/panzer/disc-fe/src/Panzer_IntrepidOrientation.cpp (hunk 1 above)
Comment (rppawlo) [reply packages/panzer/disc-fe/src/Panzer_IntrepidOrientation.cpp:179]: Because of how our test suite built the inline mesh for unit testing. Going to follow this up with some more changes to get a test to cover the corner case.

File: packages/panzer/disc-fe/src/Panzer_L2Projection.cpp (hunk 2 above)
Comment (rppawlo) [reply packages/panzer/disc-fe/src/Panzer_L2Projection.cpp:190]: I can change that. Thanks! That particular code has been in panzer for years. I just fixed the indentation as an outer loop was removed. Might be easier to review by looking at the different commits (if its not too late). The last commit just cleaned up indentation. Functional changes were first two commits.

Comment (cgcgcg) [reply packages/panzer/disc-fe/src/Panzer_L2Projection.cpp:190]: Sounds good.
